src/ImageProcessor.Web.Plugins.PostProcessor/PostProcessor.cs

```diff
@@ -41,116 +41,142 @@ internal static class PostProcessor
+                        {
+                            var remainingProcesses = processStartInfos.Count;
+                            foreach (var processStartInfo in processStartInfos)
                             {
-                                if (processResults.ExitCode == 1)
+                                // Set default properties
+                                processStartInfo.FileName = Path.Combine(postProcessorBootstrapper.WorkingPath, processStartInfo.FileName);
+                                processStartInfo.CreateNoWindow = true;
+                                processStartInfo.WindowStyle = ProcessWindowStyle.Hidden;
+
+                                // Run process
+                                using (var processResults = await ProcessEx.RunAsync(processStartInfo, cancellationTokenSource.Token).ConfigureAwait(false))
                                 {
-                                    ImageProcessorBootstrapper.Instance.Logger.Log(typeof(PostProcessor), $"Unable to post process image for request {context.Request.Unvalidated.Url}, {processStartInfo.FileName} {processStartInfo.Arguments} exited with error code 1. Original image returned.");
-                                    break;
+                                    if (processResults.ExitCode == 1)
+                                    {
+                                        ImageProcessorBootstrapper.Instance.Logger.Log(typeof(PostProcessor), $"Unable to post process image for request {context.Request.Unvalidated.Url}, {processStartInfo.FileName} {processStartInfo.Arguments} exited with error code 1. Original image returned.");
+                                        break;
+                                    }
+                                }
+
+                                remainingProcesses--;
+
+                                var destinationFileInfo = new FileInfo(destinationFile);
+                                if (destinationFileInfo.Exists)
+                                {
+                                    // Delete the source file
+                                    sourceFileInfo.IsReadOnly = false;
+                                    sourceFileInfo.Delete();
+
+                                    if (remainingProcesses > 0)
+                                    {
+                                        // Use destination file as new source (for the next process)
+                                        destinationFileInfo.MoveTo(sourceFile);
+                                    }
+
+                                    // Swap source for destination
+                                    sourceFileInfo = destinationFileInfo;
+
+                                    // Try to keep the file in memory and ensure it's not indexed
+                                    sourceFileInfo.Attributes |= FileAttributes.Temporary | FileAttributes.NotContentIndexed;
```

Comment:
I'm not sure this makes any difference, as the file is already written. It probably depends whether the file is already flushed to disk, so it doesn't hurt to set it anyway.